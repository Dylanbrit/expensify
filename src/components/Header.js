import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
    <div>
        <h1>Expensify</h1>
        <NavLink to="/">Home</NavLink>
        <NavLink to="/add">Add</NavLink>
        <NavLink to="/edit">Edit</NavLink>
        <NavLink to="/help">Help</NavLink>
    </div>
)

export default Header