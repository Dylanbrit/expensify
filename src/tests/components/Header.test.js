import { shallow } from 'enzyme'
import React from 'react'
import Header from '../../components/Header'
// react-test-renderer is a library that allows us to render components inside regular JS code and we can assert something about what got rendered
// snapshots allow us to track changes to data over time
// enzyme is a renderer for React released by Airbnb, it is a full featured renderer

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()

    // expect(wrapper.find('h1').text()).toBe('Expensify')
    // const renderer = new ReactShallowRenderer()
    // renderer.render(<Header />)
    // expect(renderer.getRenderOutput()).toMatchSnapshot()
})