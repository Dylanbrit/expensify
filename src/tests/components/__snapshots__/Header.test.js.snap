// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    to="/add"
  >
    Add
  </NavLink>
  <NavLink
    to="/edit"
  >
    Edit
  </NavLink>
  <NavLink
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
